### PR TITLE
DL-2783 ES8 for self-heal

### PR DESCRIPTION
**Pull request by jmountain-hmrc**
# DL-2783 ES8 for self-heal

**Bug fix**

* Self-heal checks from users arriving from business-customer should now perform ES8 on success

## Checklist

 - [x]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [x]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [x]  I've executed the acceptance test pack locally to ensure there are no functional regressions
 - [x]  I've added my code using logical, atomic commits, squashing as appropriate - including the JIRA issue number in the commit message
 - [x]  I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

**Comment by paulanderson1234** [comment]
Unit and integration tests run OK
AT regression test OK

File: test/connectors/AWRSConnectorSpec.scala [hunk 1]
```diff
@@ -572,54 +572,55 @@ class AWRSConnectorSpec extends AwrsUnitTestTraits {
...
-    def testCall(implicit  hc: HeaderCarrier): Future[Boolean] = testAWRSConnector.checkEtmp(testBusinessCustomer, testBusinessRegistrationDetails)
+        .thenReturn(Future.successful(HttpResponse(responseStatus, responseData)))
+    def testCall(implicit  hc: HeaderCarrier): Future[Option[SelfHealSubscriptionResponse]] = testAWRSConnector.checkEtmp(testBusinessCustomer, testBusinessRegistrationDetails)
 
-    val checkRegimeModelResponseSuccess = CheckRegimeModel(testBusinessCustomer, testBusinessRegistrationDetails)
-    val checkRegimeModelResponseSuccessJson = CheckRegimeModel.format.writes(checkRegimeModelResponseSuccess)
+    val checkRegimeModelResponseSuccess = SelfHealSubscriptionResponse("123456")
+    val checkRegimeModelResponseSuccessJson = Json.toJson(checkRegimeModelResponseSuccess)
 
     "Return true after receiving an OK response" in {
```
Comment (davidmthornton) [inline test/connectors/AWRSConnectorSpec.scala]: This is no longer Boolean - please update this

File: test/connectors/AWRSConnectorSpec.scala [hunk 2]
```diff
@@ -572,54 +572,55 @@ class AWRSConnectorSpec extends AwrsUnitTestTraits {
...
       FeatureSwitch.enable(AWRSFeatureSwitches.regimeCheck())
       mockResponse(OK, checkRegimeModelResponseSuccessJson)
-      val result: Boolean = await(testCall)
+      val result: Option[SelfHealSubscriptionResponse] = await(testCall)
 
-      result shouldBe true
+      result shouldBe Some(subscribeAcceptedResponse)
     }
 
     "Return false after receiving a NO CONTENT response" in {
```
Comment (davidmthornton) [inline test/connectors/AWRSConnectorSpec.scala]: And this...

File: test/connectors/AWRSConnectorSpec.scala [hunk 3]
```diff
@@ -572,54 +572,55 @@ class AWRSConnectorSpec extends AwrsUnitTestTraits {
...
-      mockResponse(NO_CONTENT, checkRegimeModelResponseSuccessJson)
-      val result: Boolean = await(testCall)
+      mockResponse(NO_CONTENT)
+      val result: Option[SelfHealSubscriptionResponse] = await(testCall)
 
-      result shouldBe false
+      result shouldBe None
     }
 
     "Return false after receiving a BAD REQUEST response" in {
```
Comment (davidmthornton) [inline test/connectors/AWRSConnectorSpec.scala]: And this...

File: test/connectors/AWRSConnectorSpec.scala [hunk 4]
```diff
@@ -572,54 +572,55 @@ class AWRSConnectorSpec extends AwrsUnitTestTraits {
...
-      mockResponse(BAD_REQUEST, checkRegimeModelResponseSuccessJson)
-      val result: Boolean = await(testCall)
+      mockResponse(BAD_REQUEST)
+      val result: Option[SelfHealSubscriptionResponse] = await(testCall)
 
-      result shouldBe false
+      result shouldBe None
     }
 
     "Return false after an exception has been thrown" in {
```
Comment (davidmthornton) [inline test/connectors/AWRSConnectorSpec.scala]: And this...

File: test/connectors/AWRSConnectorSpec.scala [hunk 5]
```diff
@@ -572,54 +572,55 @@ class AWRSConnectorSpec extends AwrsUnitTestTraits {
...
-      mockInvalidResponse(BAD_REQUEST, checkRegimeModelResponseSuccessJson)
-      val result: Boolean = await(testCall)
+      mockInvalidResponse(BAD_REQUEST)
+      val result: Option[SelfHealSubscriptionResponse] = await(testCall)
 
-      result shouldBe false
+      result shouldBe None
     }
+
     "Return false when feature flag is false" in {
```
Comment (davidmthornton) [inline test/connectors/AWRSConnectorSpec.scala]: And this...

**Pull request merged**